### ISSUE TITLE
Thanks for CodeKit 2 @bdkjones

### ISSUE THREAD

**Issue by Horttcore** [bdkjones/CodeKit2#236]
<pre><code>I've been super dejected lately because there's been so many issues with
the beta (which makes me think I should never touch a code editor ever again) and
virtually no one has said anything about 2.0 being good... it's just been all
bugs and issues. I'm very worried that I'm about to ship something that no one
will like, so it's nice to hear at least one person excited about it.</code></pre>

After reading this comment by @bdkjones I recognized that we never said thank you for his awesome work, making our live so much easier! CodeKit made a bigger impact on my daily work, then switching from TextMate to Sublime Text. The simple GUI (CK1) was really easy to use and I can't remember ever missing something.

CodeKit 1 showed me the stylus which I kinda felt in love for. The not-so-beloved brother in the family of SASS and LESS is for me the true hero. I'm using stylus for every project now.

After using CodeKit 2 in the beta, I don't wanna open CodeKit 1 again! In the first beta I've checked, I was pretty surprised about the new GUI, which kinda shocked me in the beginning. It was different in the start but after 10mins of using I'm used to the candy buttons ;-)

The first hassle with the new server settings and removing the browser reload function was a bit like WTF. After the WordPress fix for the server I was impressed how cool this feature really is. Awesome handy. In first I thought using Autoprefixer was a killer feature for me I can say the server is much cooler!

Maybe we didn't said anything good about CodeKit 2 yet, because all of the Warnings in the README not to do it. At least we could open an issue and tell how awesome it is here in the project ;-)

tl;dr

@bdkjones Your are rocking our worlds with CodeKit, making our daily work more enjoyable. Thanks for that! You are awesome!

![bdkjones is awesome](http://d2tq98mqfjyz2l.cloudfront.net/image_cache/1383034920531908.jpg)

**Comment by dcsturm** [comment]
Yepp, true words :)

**Comment by bdkjones** [comment]
Ha, thanks guys! I'm getting less nervous about shipping 2.0 and I really appreciate all your work and patience through the betas.

**Comment by guill3m** [comment]
Couldn't agree more.

Never jumped to CSS preprocessors until I discovered LESS.app by the awesome @bdkjones, it didn't took too long to make the jump to CodeKit, and since then has been a long time now that I don't conceive my workflow when doing front-end stuff without CodeKit.

Actually, CodeKit opened me the doors for SCSS, made my final products better (less css files, les js files since CodeKit concatenates them, all minified, etc. I didn't had any workflow like that before), and made my work easier and better. And for just $30, that's a _really good_ value.

The thing is: that was for CodeKit 1, now, with this short time with CodeKit 2, 1 feels not enough, you did it again with 2, now I can't conceive my workflow without CodeKit 2. I mean; Grunt, Gulp, Assetic, and things like that are nice for deliver the projects, but for doing the job there is no option like CodeKit.

**Thank you Bryan.**

![You're awesome Bryan, thank you.](https://i.chzbgr.com/maxW500/3898507008/hE0CE0EC1/)

**Comment by johnsardine** [comment]
Shame on us! But it's true, CodeKit 2 is awesome, when the server stuff was fixed, it was awesome, because it solves so many problems, such as testing in virtual machines, without me having to manually set the url as an ip, so that it works consistently across machines, the live changes are much better than on CodeKit 1 (it seemed that sometimes hey got caught up in a queue, and chrome was very slow at times), overall, CodeKit is fantastic, very well built, and I use it every day.

The only thing I think it could change is that it could be less noisy when it asks me to feed him in the middle of the night, other than that, it's perfect.

I can't wait for it to be shipped, I think people will genuinely be pleased and see how good it is.

Thank you

**Comment by jornki** [comment]
CodeKit for me represents the "upper" level of software. Within a front-end world with a bunch of awesome tools like Grunt, LiveReload and so on, I see CodeKit as "the next level" of application, kinda like Panics Transmit is to the ftp client world.

It is clear that a lot of care has gone into making CodeKit 2 and this is something everybody who buys it will notice. It will be bugs, but it works pretty much flawlessly for my daily work as of now..including the server stuff. It's already a great product - before it has been released.

**Comment by rpkoller** [comment]
Well Bryan you shouldn't be worried about 2.0, the amount of bugs you've mentioned in the betas  as well as the shipping. CodeKit is a complex approach with many many many tools and frameworks included - each of them have their very own quirks, bugs and flaws - plenty of them. 
Taken that in consideration CodeKit 2 is rock solid and the number of bugs found in the beta phase is rather moderate and small  compared to CodeKits complexity (i would have expected much more). I guess non of the beta testers would have spend their time for bug hunt if CodeKit wouldn't be worth it. Why invest time in a software mediocre, weak and or in question. It's about making a rock solid app even more polished, reliable and outstanding for the everyday use.

**Comment by newelement** [comment]
CodeKit 2 is awesome. I couldn't imagine the amount of time that went in to make it. All of our issues just makes it even better. I'm ready to throw my money at it.

**Comment by coreyallen** [comment]
I've been doing software QA for almost ten years now...crap I gotta switch careers...where was I?  Oh yeah.  I've never experienced a Beta that has been so stable and had such a trajectory of improvement from the start.  I've been using it on all of my freelance gigs and it's been invaluable the whole way. 

Thanks for making my side job easier.

**Comment by devilskitchen** [comment]
I'd like to add my voice to the chorus of approval. As someone said near the top of the thread, I never thought that I needed anything added to CK1: I honestly didn't see how it could be improved.

But I've now got it running and Auto-prefixer is really cool—really cool.

But the best thing is the new server functionality, allowing for remote reloads—I am now sitting surrounded by two screens (I couldn't see the point in more than one before CK2) and my iPad and my iPhone, and a colleague's Android phone—all reloading my web application as I build it.

Seriously, @bdkjones, CK2 is utterly awesome and I feel privileged to get the chance to use it early on, and provide feedback towards making it better.

**Comment by rasmusbe** [comment]
I couldn't agree more!

CK1 made me use LESS and JSHint, the betas of CK2 has already made me try new techniques (SCSS, Autoprefixer, Source maps) and I'm probably going to try out even more things since CK makes it so easy to use.

So thank you, and please take my money!

**Comment by zengabor** [comment]
Yes, please take my money. I am definitely going to go for a new license instead of an upgrade and choose the highest price.

**Comment by donaldallen** [comment]
CodeKit 2 will be in every web developer's toolkit. Guaranteed. It's that good. @bdkjones, thank you for letting me participate in the beta.

**Comment by shshaw** [comment]
+1. Can definitely reproduce the "CodeKit 2 is awesome" issue every time I use the app.

**Comment by greystate** [comment]
[√] All of the above

I have been using TextMate 1 and CodeKit 1 (Less.app before CK was a thing) for very long — recently switched to TextMate 2 and how appropriately that CodeKit gets a version bump :) They're in sync again!
I've had loads of freelance gigs where the teams were using everything from a myriad of Visual Studio plugins & grunt/gulp to plethoras of "require this js and that js" etc., and I can't tell you how many HOURS I've wasted waiting for those tools to "be reconfigured because of the X, Y & Z" — and every single time, I knew MY code was clean and ready because of CodeKit & TextMate not having to be configured in any way special - they just frickin' work.

I love this combo - and @bdkjones: I absolutely love this app - it has made a world of difference to me, and I am very sorry for being too scared at the "you don't talk about ~~Fight Club~~CodeKit :)" intro, to take the time to express how I feel about it. You rock!

**Comment by jensimmons** [comment]
Yay! I'm glad @Horttcore opened this issue. Thanks Ralf. 

Bryan, you did scare us silent with your warnings. And good thing, because I would have tweeted "CODEKIT2 IS AWESOME!!!!!!!!!!!!! \o/" at least a dozen times by now if you hadn't pled such a strong case for waiting. 

I totally know the feeling of being alone working on something, feeling like it's not really that great because nothing is happening publicly and no one seems to care. It's not true, through. It's just a weird artifact of 21 century life, sitting alone at a desk typing bits. You are making something, and that something is really great. If we beta testers all see it, I can't imagine the wider public won't see it immediately as well. 

I've been using CodeKit 1 every day I have a code editor open, ever since I started using Sass (and discovered that running the ruby command from the CLI melts my three-year-old Mac Book Air). It's a requirement to my workflow — Coda + Git Tower + Codekit. On. every. project. 

The new features of CK2 are just making work life that much more fluid and fun. Thank you!

**Comment by hartless** [comment]
Totally agree. CodeKit got me started in Sass. CK2 has made me build better and faster thanks to the built-in server, live refresh and the amazing Bower integration. Having a single development URL now keeps my sanity intact when dealing with multiple projects.  

I kept my comments brief and to the point when filing issues out of respect for your time. That said, this new version totally rocks! If it weren't for CodeKit, I'd probably quit my job and become an accountant. And who the hell wants that.

Kudos to you and Guy for this.

**Comment by sethta** [comment]
:+1: CodeKit was amazing and I wondered why I didn't use it sooner. Now with CK2, it's really hard to go back to CK. I know because I dropped my MacBook (yes, I cried a little) and had to get it repaired. My old MacBook had CK installed on it and when you go back after using CK2 for a while, there's a HUGE difference, especially with the speed of compiling and the ability to use phones and tablets when editing. Needless to say, after using CK1 for a couple hours, I had to switch to CK2.

Thanks for a great product, and even better support. The WordPress caching bug with Chrome is a great example. You could have just said "Use Safari" but you spent the time and brainpower to figure it all out and I thank you for that. :grinning:

**Comment by kmmathis** [comment]
Dude... I LOVE CODEKIT 2. There's a hundred things I can say about the app that heighten my enjoyment of life, as well as improve the quality of my work. This is an amazing asset to the web development community; I'm sure many of us would still be writing plain CSS and manually refreshing the browser (like chumps) if CodeKit didn't exist.

I hope that when you ship CK2, you let us specify a larger payment amount than what you're asking for. I'm well aware of the value of this software, and I'm prepared to pay for it.

Thanks for your hard work and good ideas. Much appreciated!

**Comment by mkoistinen** [comment]
Wow. Just when you think CK couldn't get any More Awesomer™, CK2 shows you how its done.

I love how it anticipates my needs!

**Comment by BabylonSix** [comment]
Bryan. THANK YOU for making CK2!

CK gave me the courage to actually make the jump from READING about doing Web Development, to actually doing it, with awesome results to boot. It makes managing complexity a non-issue with stylus and jade, I absolutely loved the fact that the feedback loop between writing something and seeing the results was so small.

With CK2, I have all the advantages of one, but now I can actually do responsive design without second guessing how things will look like on smaller screens. I can actually target a screen size and work with it, without getting lost.

Prefixer and Nib, and the inclusion of bower just make for another world of pain go away, even more complexity managed! I mean how much better can this thing get?

Bryan, don't stop, and don't lose spirit,
you're changing the world in a VERY REAL way, and for that THANK YOU! ^_^

**Comment by jonnitto** [comment]
Yep. The same here. Codekit changed the way I work. Codekit 2 will push this even further! Thank you so much for this masterpiece of software!

**Comment by jeremysexton** [comment]
![](http://replygif.net/i/716.gif)

**Comment by JesusGarciaValadez** [comment]
Yes. It's true. CK2 it's a great piece of software. And none of us appreciate it as we should. Bryan, thanks for taking your time for make our work more fun, easy and reliable with your work and comments.

**Comment by peterhartree** [comment]
+100 to all of the above.

**Comment by jslaterTO** [comment]
Totally agree with everyone else here.  After working exclusively with CK2 more or less since you started releasing the betas, I couldn't imagine going back to CK1, let alone not working with it at all.

**Comment by pixeljitsu** [comment]
Just take a moment and think of how many barriers you will/have removed for developers who are not already familiar with the CLI. Not everyone one is going to feel at home with guard and livereload, and even for those of us that are, CK2 is simply way less of a hassle anyway.  

Thank you for all the time you have saved me personally by including Bower and all the other great and well thought out features in CK2. I would rather spend my time actually coding than worrying about esoteric gem dependencies and brew versions.

What you have essentially created is a tool that will save all of us significant amounts of time and effort, and that means greater profits and more time to spend however we wish.

Your efforts and dedication have certainly garnered my loyalty. I believe that CK2 will be a huge success.

Thank you!

**Comment by geebru** [comment]
Even more love - you've taken an incredible app and multiplied it tenfold. CK2 is absolutely astounding and I've yet to find something I wish it did missing while continually finding new things that blow me away.

Keep up the awesome work!

Thank you!

**Comment by frank-laemmer** [comment]
yes, can reproduce CK2 is awesome. enjoying the smart mobile browser auto refresh via Bonjour.

**Comment by yratof** [comment]
I've got to agree, my workflow is on heroin with CK2. And sat in a studio with two other guys using CK1, and i'm whizzzing through things at a phenomenal pace, and they're getting on like usual. In fact, I recall yesterday, Tom saying "Codekit is really clever when you think about how much it does for you in no time at all", I can imagine when he gets his hands on codekit 2, he'll have a much bigger statement to say in regards to speed and what it'll do for him.

If there's a point where we can get a google hangout (if that even exists) of all the beta testers, we could all have a 'cold one' together :)

**Comment by devilskitchen** [comment]
I feel a CK2 beta-testers boozy Friday coming on...

**Comment by bdkjones** [comment]
Thanks again for the support! 

Beta 19 is damn close. I've got the website 90% done. Going to start on the screencasts this weekend. 

If any of you are going to be in San Diego on 8 March, let me know. A few friends and I are going to go celebrate.

**Comment by rpkoller** [comment]
I've already written that i dig ck2 a few posts above; but a few minutes ago I've tried one thing out of curiosity. Set up port forwarding in my router for the computer running ck2 and then got my external ip via whatismyip.com and surfed via safari to the ip with the ck2 port at the end. The site showed up but then I've edited the scss of the site displayed, saved it and then got live reloaded. WOW I haven't expected that, guessed it would work only in the local network! Mighty mighty mighty! Maximum cudos Bryan!!!

**Comment by bdkjones** [comment]
Ha. Now that's pretty awesome. Could be a cool way to show off the site for clients.

**Comment by bdkjones** [comment]
Ralph - I'd like to test this. 

When would be a good time to do so? I'd like to visit your IP address on my Mac over here and have you make some changes to your project to see if the live-reloads come through across the Internet. 

Please let me know what timezone you're in and when you're available. Thanks.

**Comment by rpkoller** [comment]
I'll write you a mail with the ip and necessary infos.
